homeassistant/components/modbus/modbus.py

```diff
@@ -158,12 +158,10 @@ async def async_modbus_setup(
 
     async def async_stop_modbus(event: Event) -> None:
         """Stop Modbus service."""
-
-        async_dispatcher_send(hass, SIGNAL_STOP_ENTITY)
         for client in hub_collect.values():
             await client.async_close()
 
-    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_stop_modbus)
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_CLOSE, async_stop_modbus)
```

Comment:
why do we need to change from STOP? Integrations don't usually have to use CLOSE.

Reply:
The restore dump happens in the stop event, so the state was not saved because the entity states where removed already. 
Imo the whole thing can be removed because HA is restarting so all connections will close.

Reply:
Isn't the problem that we changed the entity state to unavailable when stopping modbus via the `SIGNAL_STOP_ENTITY` signal. Then the correct state won't be stored when dumping state for the restore state feature. If we've removed calling that signal, we shouldn't need to change the event we listen for here.

Reply:
I'm not 100% sure if we tested that or not @crug80, can you check when you have time what happens if you change back to stop instead?

Reply:
@VandeurenGlenn not problem to test it, but i didn't understand the scenario. I restore the old line 164 with the EVENT_HOMEASSISTANT_STOP instead of CLOSE?

Reply:
Tested the PR with only  the line 162 REMOVED, keeping the rest of code as it with EVENT_HOMEASSISTANT_STOP and it works. The state is correctly stored. Someone can explain me the reason behind? :)

Reply:
@crug80 perfect, line 162 makes it so that all of the integration entities get stopped before the actual stop event runs in core where the dump happens.
In short the ’SIGNAL_STOP_ENTITY’ sets the states to unknown.